app/classifier/tasks/survey/index.cjsx

```diff
@@ -48,7 +48,8 @@ module.exports = React.createClass
       {if @state.selectedChoiceID is ''
         <Chooser task={@props.task} filters={@state.filters} onFilter={@handleFilter} onChoose={@handleChoice} onRemove={@handleRemove} annotation={@props.annotation} focusedChoice={@state.focusedChoice} />
       else
-        <Choice task={@props.task} choiceID={@state.selectedChoiceID} onSwitch={@handleChoice} onCancel={@clearSelection} onConfirm={@handleAnnotation} />}
+        existingAnnotation = _.find(@props.annotation.value, 'choice', @state.selectedChoiceID) || {}
+        <Choice annotation={existingAnnotation} task={@props.task} choiceID={@state.selectedChoiceID} onSwitch={@handleChoice} onCancel={@clearSelection} onConfirm={@handleAnnotation} />}
```

Comment:
`@handleAnnotation` is causing the bug where each choice is pushed to the end of the `values` array as a new annotation, even if that choice already exists.